### DISCUSSION TITLE
点击事件穿透问题

### DISCUSSION THREAD

**Discussion by mc15829702687** [zouyaoji/vue-cesium#268] [Q&A] [answered]
大佬，实体点击事件穿透到视图点击事件怎么解决

**Comment by zouyaoji** [comment, marked as answer]
意思是实体和viewer都响应了点击事件么？应该是正常的哈。

请提供代码再进一步讨论哈

**Reply by mc15829702687** [reply]
<vc-viewer @left-click="onLeftClick">
    <vc-entity @mousedown="onmousedown"></vc-entity>
 </vc-viewer>
 
 大佬，类似上面代码，会先走mousedowm事件，然后再走viewer的点击事件，我想在实体mousedown时阻止viewer的click事件，请问有什么好办法吗，我试过.stop修饰符，直接报错

**Reply by zouyaoji** [reply]
暂时没有想到好的办法。

具体是什么需求需要这样啊？

**Reply by mc15829702687** [reply]
大佬，需求类似于图上打点，当点击entity显示详情框，当点击实体外部，详情框隐藏

然后，我看大佬写的实体组件有个clickout事件，但是也会触发viewer组件的left-click事件，详情框会显示后瞬间隐藏

**Reply by zouyaoji** [reply]
那这就简单了哈，用这个组件就行[vc-selection-indicator](https://zouyaoji.top/vue-cesium/#/zh-CN/component/controls/vc-selection-indicator)

**Reply by mc15829702687** [reply]
谢谢大佬，我看下

**Reply by zouyaoji** [reply]
能行吧？

**Reply by mc15829702687** [reply]
可以，谢谢大佬，百忙之中帮忙解决问题

**Reply by zouyaoji** [reply]
小事情
